src/iop/channelmixerrgb.c

```diff
@@ -3855,12 +3855,8 @@ void reload_defaults(dt_iop_module_t *module)
     }
 
     if(dt_image_is_matrix_correction_supported(img) && !dt_image_is_monochrome(img))
-    {
-      if(dt_bauhaus_combobox_length(g->illuminant) < DT_ILLUMINANT_CAMERA + 1)
-        dt_bauhaus_combobox_add_full(g->illuminant, _("as shot in camera"),
-                                     DT_BAUHAUS_COMBOBOX_ALIGN_RIGHT,
-                                     GINT_TO_POINTER(DT_ILLUMINANT_CAMERA), NULL, TRUE);
-    }
+      // add back if needed
+      dt_bauhaus_combobox_set_from_value(g->illuminant, DT_ILLUMINANT_CAMERA);
```

Comment:
`gui_update` and `dt_bauhaus_update_module` will be called after `reload_defaults` so that's where the value will be set to the user selected option.

I could explicitly mention that here, but I always struggle with how much commentary to put into the code base (since often we start from nothing). Putting the first paragraph in a comment in the code might lead to a situation where somebody changes the control flow and this is actually no longer true, but not updating comments and thereby creating more confusion. So in the end one needs to know quite a bit about the gui update processes to understand what is safe to touch. It took me a while...

BTW in this case an update to the combo value doesn't cascade to possible dependent processes because during `reload_defaults` `darktable.gui->reset` is TRUE.

Reply:
Fair enough.

Just expanding on this, would this warrant a `dt_bauhaus_combobox_add_from_value` which just adds the item from introspection (if missing) and does not change the actual set value? This would communicate the intent better here, removing the need for a separate explanatory comment.

Reply:
Then I might as well use `dt_bauhaus_combobox_add_introspection` which is more verbose but also more efficient.
I will need to add a check if the value currently is available (and return its position) and refactor a little using this `dt_bauhaus_combobox_get_from_value`; that will also make removing items (in gui_init) safer.

Reply:
@flannelhead Last commit not exhaustively tested yet, but would like your feedback on whether this actually improves readability or not...

Reply:
@dterrahe I left some comments now, let's see what you think of them. I'll let @TurboGit also comment on the readability side.